include/meltpooldg/heat/heat_transfer_operator.hpp

```diff
@@ -827,6 +854,20 @@ namespace MeltPoolDG::Heat
                      temp_vals.get_value(q_index);
             }
 
+          if (velocity && data.solidification)
+            {
+              get_material_parameter_second_derivatives_with_solidification(
+                d2_capacity_dT2, d2_density_dT2, temp_lin_vals.get_value(q_index));
+              val += (d_capacity_dT * density + d_density_dT * capacity) *
+                     (2.0 * temp_lin_vals.get_value(q_index) * temp_vals.get_gradient(q_index) +
+                      temp_vals.get_value(q_index) * temp_lin_vals.get_gradient(q_index)) *
+                     velocity_vals.get_value(q_index);
```

Comment:
Couldn't you simply omit the second line?
```suggestion
                     (2.0 * temp_lin_vals.get_value(q_index) * temp_vals.get_gradient(q_index) *
                     velocity_vals.get_value(q_index);
```

Reply:
When deriving the consistent tangent from the residual, the product rule results in two identical terms. Here I used the factor 2 instead of writing down the term twice. I already updated the Theory Manual if you want to to have a look at the derivation.

Reply:
Okay, now I can follow :-). The expression should be correct.